feat: basic proposal for trajectory (#191)
i-am-bee/acp: docs/core-concepts/message-structure.mdx

## Message Parts

| Attribute                  | Required | Description                                                  |
| :------------------------- | :------- | :----------------------------------------------------------- |
| `content_type`             | Yes      | MIME type (e.g., `text/plain`, `image/png`)                  |
| `content` OR `content_url` | Yes      | Content `inline` or via `URL`                                |
| `content_encoding`         | No       | `"plain"`  (default) or `"base64"`                           |
| `name`                     | No       | Makes this part an **Artifact**                              |

> | `metadata`                 | No       | Additional metadata like citations or trajectory information |

Author: ```suggestion
| `metadata`                 | No       | Additional metadata to provide additional context or semantic information. |
```

just to avoid maintenence.

> | `metadata`                 | No       | Additional metadata to provide additional context or semantic information. |

Reviewer: ```suggestion
| `metadata`                 | No       | Additional metadata to provide additional context or semantic information |
```

- Files
- Named results

## Metadata

MessageParts can include optional `metadata` to provide additional context or semantic information. ACP supports two types of metadata:

### Citation Metadata

> Used to reference information sources and provide attribution for content. Citations can be rendered as inline icons or text range annotations.

Author: ```suggestion
Used to reference information sources and provide attribution for content. Citations are defined as range annotations in the generated content..
```

To avoid coupling with "UI".

**Properties:**
- `url`: URL of the source document
- `title`: Title of the source document  
- `description`: Description or snippet from the source
- `start_index` / `end_index`: Character range for text citations

### Trajectory Metadata

> Used to capture an agent's reasoning steps or tool execution information, providing transparency into the agent's decision-making process.

Author: ```suggestion
Used to provide transparency into the agent's decision-making process; For example, reasoning steps or tool execution information.
```

Reviewer 2: I wouldn't use the term "decision-making process" here.  The output of a tool execution isn't a decision taken by the agent, it's just the output of the tool.  Perhaps this:

> Used to provide transparency into the agent's operation.  For example, logging decisions made, steps taken, tools called, reasoning made, or any other operational information the agent wishes to provide.

Author: The reason why i'd prefer the "decision making" is just because it's up to the agent to decide whether the tool is called or not. In other words just knowing tool has(not) been called is valuable information by itself. 

I understand that tool output and input does not have anything related to the actual decision making, but that's something i perceive as an additional metadata - i believe just knowing whether tool has been called or not is the primary trajectory information.

Also the term trajectory itself seems very synonymous to "decision making".
